### PR TITLE
Bump dependencies and fix errors

### PR DESCRIPTION
**Pull request by bolt12**

### CODE REVIEW

File: src/System/Metrics/Network/Acceptor.hs [hunk 1]
```diff
@@ -79,6 +79,7 @@ doListenToForwarder snocket address timeLimits app = do
   _ <- async $ cleanNetworkMutableState networkState
   withServerNode
     snocket
+    (\_ _ -> return ())
```
Comment (coot) [inline src/System/Metrics/Network/Acceptor.hs:82]: @bolt12 I think we don't need to expose this argument at all for `withServerNode`, that's runs non P2P server anyway.

File: src/System/Metrics/Network/Forwarder.hs [hunk 2]
```diff
@@ -68,6 +68,7 @@ doConnectToAcceptor
 doConnectToAcceptor snocket address timeLimits app =
   connectToNode
     snocket
+    (\_ -> return ())
```
Comment (coot) [inline src/System/Metrics/Network/Forwarder.hs:71]: The same here.

**Review by coot — APPROVED** [review]
The haskell part looks good to me.  I haven't reviewed the nix part, could you just say why it's needed?

**Comment by coot** [comment]
@bolt12 it's ok to include if you think so!

**Comment by bolt12** [comment]
bors mergee

**Comment by bolt12** [comment]
bors merge

**Comment by bolt12** [comment]
@coot I don't think I can merge this

**Review by coot — APPROVED** [review]

**Comment by michaelpj** [comment]
It would be great if we could get lower-bounds in the cabal files for the versions that we know no longer work because you had to make changes to get the new ones to compile.

**Pull request merged**